### ISSUE TITLE
`allowEmpty={false}` for related fields that are also flagged as required

### ISSUE THREAD

**Issue by cobusc** [DavidOberholzer/SwaggerReactAdminGenerator#3]
Given a model defined in the Swagger `definitions` section which has a field with an `x-related-info` annotation, the following code is currently being generated.
```
                        <ReferenceField
                            label="Deletion Method"
                            source="deletion_method_id"
                            reference="deletionmethods"
                            sortable={false}
                            linkType="show"
                            allowEmpty
                        >
                            <NumberField source="label" />
                        </ReferenceField>
```
This does not take into consideration whether the field is also `required`, in which case `allowEmpty` must be false, like this:
```
                        <ReferenceField
                            label="Deletion Method"
                            source="deletion_method_id"
                            reference="deletionmethods"
                            sortable={false}
                            linkType="show"
                            allowEmpty={false}
                        >
                            <NumberField source="label" />
                        </ReferenceField>
```

**Comment by DavidOberholzer** [comment]
The `allowEmpty` prop only applies to the `Input` select type fields in React-Admin. But will make this change to the inputs

**Comment by DavidOberholzer** [comment]
Fixed in #4
